File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    float letras = count_letters(text);
    float palavras = count_words(text);
    float frases = count_sentences(text);

    int conta1 = (letras / palavras) * 100;
    int conta2 = (frases / palavras) * 100;

    float index = ((0.0588 * ((letras / palavras) * 100)) - (0.296 * ((frases / palavras) * 100))) - 15.8;
    int indexc = round(index);

    if (indexc < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (indexc > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", indexc);
    }
}

int count_letters(string text)
{
    int cont = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] != ' ' && text[i] != '.' && text[i] != ',' && text[i] != '!' && text[i] != '?' && text[i] != ':' && text[i] != '\"' && text[i] != ';' && text[i] != '-' && text[i] != '\'')
        {
            cont++;
        }
    }
    return cont;
}

int count_words(string text)
{
    int cont = 1;

    for (int j = 0, len = strlen(text); j < len; j++)
    {
        if (text[j] == ' ')
        {
            cont++;
        }
    }

    return cont;
}

int count_sentences(string text)
{
    int cont = 0;

    for (int k = 0, len = strlen(text); k < len; k++)
    {
        if (text[k] == '.' || text[k] == '!' || text[k] == '?')
        {
            cont++;
        }
    }
    return cont;
}

File: sentimental-readability/readability.py
def count_letters(text):
    special_characters = ".,!?:;'-\""
    count = sum(1 for char in text if char.isalpha() and char not in special_characters)
    return count

def count_words(text):
    return len(text.split())

def count_sentences(text):
    sentence_endings = ".!?"
    count = sum(1 for char in text if char in sentence_endings)
    return count

def main():
    text = input("Text: ")
    letras = count_letters(text)
    palavras = count_words(text)
    frases = count_sentences(text)

    conta1 = (letras / palavras) * 100
    conta2 = (frases / palavras) * 100

    index = ((0.0588 * conta1) - (0.296 * conta2)) - 15.8
    indexc = round(index)

    if indexc < 1:
        print("Before Grade 1")
    elif indexc > 15:
        print("Grade 16+")
    else:
        print(f"Grade {indexc}")

if __name__ == "__main__":
    main()
